File: client-leger/app/src/main/java/interfaces/dto/createMatchDto.kt
package interfaces.dto

import com.auth0.androidlogin.models.IGame
import com.google.gson.annotations.SerializedName

data class CreateMatchDto (
    @SerializedName("game")
    val game: IGame,

    @SerializedName("managerName")
    val managerName: String,

    @SerializedName("managerId")
    val managerId: String,

    @SerializedName("isTeamMatch")
    val isTeamMatch: Boolean,

    @SerializedName("isPricedMatch")
    val isPricedMatch: Boolean,

    @SerializedName("isFriendMatch")
    val isFriendMatch: Boolean,

    @SerializedName("priceMatch")
    val priceMatch: Number,

)

File: server/app/classes/match/dto/createMatchDto.ts
import { Player } from '@app/interfaces/player';
import { Game } from '@app/model/database/game';

export interface CreateMatchDto {
    game: Game;
    managerName: string;
    managerId: string;
    isFriendMatch: boolean;
    isTeamMatch: boolean;
    isPricedMatch: boolean;
    nbPlayersJoined: number;
    players: Player[];
    priceMatch: number;
}
